core/system_impl.cpp

```diff
@@ -341,6 +366,37 @@ size_t SystemImpl::total_components() const
     return _components.size();
 }
 
+void SystemImpl::register_component_discovered_callback(discover_callback_t callback)
+{
+    component_discovered_callback = callback;
+
+    if (total_components() > 0) {
+        ComponentType component_type;
+        for (const auto &elem : _components) {
+            switch (elem) {
+                case MAV_COMP_ID_AUTOPILOT1:
+                    component_type = AUTOPILOT;
+                    break;
+                case MAV_COMP_ID_CAMERA:
+                case MAV_COMP_ID_CAMERA2:
+                case MAV_COMP_ID_CAMERA3:
+                case MAV_COMP_ID_CAMERA4:
```

Comment:
Next time we'll have to refactor this switch case into a function.

Reply:
Oh, just seing it after my comment above. So that is duplicated from above, and should become something like `const auto component_type = getComponentType(component_id);`.